qubes/tests/integ/vm_qrexec_gui.py

```diff
@@ -664,6 +664,34 @@ def test_110_qrexec_filecopy_deny(self):
                 'test -d /home/user/QubesIncoming/{}'.format(
                     self.testvm1.name)))
 
+    def test_115_qrexec_filecopy_no_agent(self):
+        # The operation should not hang when qrexec-agent is down on target
+        # machine, see QubesOS/qubes-issues#5347.
+
+        self.loop.run_until_complete(asyncio.wait([
+            self.testvm1.start(),
+            self.testvm2.start()]))
+
+        with self.qrexec_policy('qubes.Filecopy', self.testvm1, self.testvm2):
+            # Make sure qrexec-agent is running before killing it
+            self.loop.run_until_complete(asyncio.wait([
+                self.wait_for_session(self.testvm2)]))
+            self.loop.run_until_complete(
+                self.testvm2.run_for_stdio('sudo pkill qrexec-agent'))
```

Comment:
Please use `user='root'` keyword argument instead of sudo, it may be not configured (for example in minimal template). It may be also better to use `systemctl stop ...` instead of killing (if we ever add automatic restart there),